src/vaults/VaultInternal.ts

```diff
@@ -283,8 +293,16 @@ class VaultInternal {
   protected async start_(
     fresh: boolean,
     tran: DBTransaction,
+    ctx?: Partial<ContextTimed>,
     vaultName?: VaultName,
-  ) {
+  ): Promise<void>;
+  @timedCancellable(true)
+  protected async start_(
+    fresh: boolean,
+    tran: DBTransaction,
+    @context ctx: ContextTimed,
+    vaultName?: VaultName,
+  ): Promise<void> {
```

Comment:
Here, as `tran` is expected to be present, and an overload also expects `ctx` to be present, the `vaultName` is the only optional parameter. Thus, it has been pushed back to the end of the argument list. This does not follow how the rest of the project has been formatted, but there is little that can be done here.

However, I am curious as to the reason for keeping the vault name optional. Does this mean that vaults can exist without vault names? I guess it would make sense, as a vault name is merely an alias and the vault ID is the actual identifier. But I'd still want to know the actual reason.

As a workaround, I can pack other parameters inside a JSON object and pass the context and transaction identifier outside like regular parameters.

Reply:
Yea vault names are optional. They can be reset, or they can be cleared.

Reply:
The signature might be wrong.